src/datumaro/plugins/data_formats/kaggle/base.py

```diff
@@ -0,0 +1,393 @@
+
+        self._path = path
+        self._columns = columns
+
+        self._items, label_cat = self._load_items(ann_file, columns)
+        self._categories = {AnnotationType.label: label_cat}
+
+    def _load_items(self, ann_file: str, columns: Dict[str, Union[str, list]]):
+        df = pd.read_csv(ann_file, header=None, on_bad_lines="skip")
+
+        indices = {}
+        for key, field in columns.items():
+            if key == "bbox":
+                indices[key] = []
+                for v in field:
+                    indices[key].append(list(df.iloc[0]).index(v))
+            else:
+                indices[key] = list(df.iloc[0]).index(field)
+
+        label_cat = LabelCategories()
+        items = []
+        for ind, row in df.iterrows():
+            if ind == 0:
+                continue
+
+            data_info = list(row)
+            media_name = data_info[indices["media"]]
+
+            if not media_name.lower().endswith(tuple(IMAGE_EXTENSIONS)):
+                for ext in IMAGE_EXTENSIONS:
+                    if osp.exists(os.path.join(self._path, media_name + ext)):
+                        media_name += ext
+                        break
+
+            label_name = "default"
+            if "label" in indices:
+                label_name = str(data_info[indices["label"]])
+            label_cat.add(label_name)
+            label, _ = label_cat.find(label_name)
+
+            if osp.exists(osp.join(self._path, media_name)):
+                items.append(
+                    DatasetItem(
+                        id=osp.splitext(media_name)[0],
+                        subset=self._subset,
+                        media=Image.from_file(path=osp.join(self._path, media_name)),
+                        annotations=[
+                            Label(label=label),
+                        ],
+                    )
+                )
+        return items, label_cat
+
+    def categories(self):
+        return self._categories
+
+    def __iter__(self):
+        yield from self._items
+
+
+class KaggleImageTxtBase(DatasetBase):
+    def __init__(
+        self,
+        path: str,
+        ann_file: str,
+        columns: Dict[str, int],
+        *,
+        subset: Optional[str] = DEFAULT_SUBSET_NAME,
+        ctx: Optional[ImportContext] = None,
+    ):
+        super().__init__(ctx=ctx)
+
+        self._subset = subset
+
+        self._path = path
+        self._columns = columns
+
+        self._items, label_cat = self._load_items(ann_file, columns)
+        self._categories = {AnnotationType.label: label_cat}
+
+    def _load_items(self, ann_file: str, columns: Dict[str, Union[int, list]]):
+        label_cat = LabelCategories()
+
+        ids = []
+        items = []
+        with open(ann_file, "r", encoding="utf-8") as f:
+            for line in f:
+                line = line.split()
+                media_name = line[columns["media"]]
+
+                if not osp.exists(osp.join(self._path, media_name)):
```

Comment:
Why don't we append a file extension from `IMAGE_EXTENSIONS` this time?

Reply:
There is no case omitting an image extension in Kaggle data. But I think it would be good to automatic detection of image extension here too. I will update.

Reply:
fixed at 0b13690c5f9a0ad5bcf55e3d6b3baf17ed84c7eb